tests/unit/reco_utils/evaluation/test_spark_evaluation.py

```diff
@@ -308,3 +373,119 @@ def test_spark_python_match(python_data, spark):
     assert map_at_k(df_true, df_pred, k=10) == pytest.approx(
         eval_spark4.map_at_k(), TOL
     )
+
+
+@pytest.mark.spark
+def test_catalog_coverage(spark_diversity_data, target_metrics):
+    train_df, reco_df = spark_diversity_data
+    evaluator = SparkDiversityEvaluation(
+        train_df=train_df, reco_df=reco_df, col_user="UserId", col_item="ItemId"
+    )
+    c_coverage = evaluator.catalog_coverage()
+    assert c_coverage == target_metrics["c_coverage"]
+
+
+@pytest.mark.spark
+def test_distributional_coverage(spark_diversity_data, target_metrics):
+    train_df, reco_df = spark_diversity_data
+    evaluator = SparkDiversityEvaluation(
+        train_df=train_df, reco_df=reco_df, col_user="UserId", col_item="ItemId"
+    )
+    d_coverage = evaluator.distributional_coverage()
+    assert d_coverage == target_metrics["d_coverage"]
+
+
+@pytest.mark.spark
+def test_item_novelty(spark_diversity_data, target_metrics):
+    train_df, reco_df = spark_diversity_data
+    evaluator = SparkDiversityEvaluation(
+        train_df=train_df, reco_df=reco_df, col_user="UserId", col_item="ItemId"
+    )
+    actual = evaluator.historical_item_novelty().toPandas()
+    assert_frame_equal(
+        target_metrics["item_novelty"], actual, check_exact=False, check_less_precise=4
+    )
+
+
+@pytest.mark.spark
+def test_user_novelty(spark_diversity_data, target_metrics):
+    train_df, reco_df = spark_diversity_data
+    evaluator = SparkDiversityEvaluation(
+        train_df=train_df, reco_df=reco_df, col_user="UserId", col_item="ItemId"
+    )
+    actual = evaluator.historical_user_novelty().toPandas()
+    assert_frame_equal(
+        target_metrics["user_novelty"], actual, check_exact=False, check_less_precise=4
+    )
```

Comment:
I think it would be good to check the limits of all these metrics. For example in [mae](https://github.com/microsoft/recommenders/blob/main/tests/unit/reco_utils/evaluation/test_python_evaluation.py#L161) we have that if the prediction and true vectors are the same, then mae is 0. I guess that if all the recommendations are the same, there is no novelty.

Reply:
Good point. For some of these there are no upper limits e.g. novelty can approach infinity.